game/Assets/Scripts/RevisionTextController.cs

```diff
@@ -0,0 +1,146 @@
+        currentTopicTranslations = new List<string>(topicTranslationDict.Keys);
+    }
+
+    /*
+     * Calculates the next vocabularyWordIndex and vocabularyTopicIndex then
+     * selects the appropriate french and english words from the vocabularyDict.
+     * If at the end of the vocabularyDict then returns null.
+     */
+    private List<string> GetNextWordPair()
+    {
+        // Check if the current index is the last entry in the entire vocab dictionary
+        int maxIndexOfFinalTopic = vocabularyDict[topics[topics.Count - 1]].Count - 1;
+        if ((vocabularyTopicIndex == topics.Count - 1) && 
+            (vocabularyWordIndex == maxIndexOfFinalTopic))
+        {
+            return null;
+        }
+
+        // Check if the current index is the last entry in the current vocab topic
+        int maxIndexOfCurrentTopic = vocabularyDict[topics[vocabularyTopicIndex]].Count - 1;
+        if ((vocabularyWordIndex == maxIndexOfCurrentTopic) &&
+            (vocabularyTopicIndex < topics.Count - 1))
+        {
+            vocabularyTopicIndex++;
+            vocabularyWordIndex = -1;
+            UpdateCurrentTopic();
+        }
+
+        vocabularyWordIndex++;
+        string currentFrenchWord = currentTopicTranslations[vocabularyWordIndex];
+        string currentEnglishWord = topicTranslationDict[currentFrenchWord];
+
+        return new List<string> { currentEnglishWord, currentFrenchWord };
+    }
+
+    /*
+     * Calculates the previous vocabularyWordIndex and vocabularyTopicIndex then
+     * selects the appropriate french and english words from the vocabularyDict.
+     * If at the start of the vocabularyDict then returns null.
+     */
+    private List<string> GetPrevWordPair()
+    {
+
+        // Check if the current index is the first entry in the entire vocab dictionary
+        if ((vocabularyTopicIndex == 0) && (vocabularyWordIndex == 0))
+        {
+            return null;
+        }
+
+        // Check if the current index is the first entry in the current vocab topic
+        if ((vocabularyWordIndex <= 0) && (vocabularyTopicIndex > 0))
+        {
+            vocabularyTopicIndex--;
+            string newTopic = topics[vocabularyTopicIndex];
+            vocabularyWordIndex = (vocabularyDict[newTopic]).Count;
+            UpdateCurrentTopic();
+        }
+
+        vocabularyWordIndex--;
+        string currentFrenchWord = currentTopicTranslations[vocabularyWordIndex];
+        string currentEnglishWord = topicTranslationDict[currentFrenchWord];
+
+        return new List<string> { currentEnglishWord, currentFrenchWord };
+    }
+
+    /*
+     * Updates the text displayed by the englishText and frenchText components
+     * to be that of the next entry in the vocabularyDict if one exists
+     */
+    public void NextCard()
+    {
+        List<string> currentPair = GetNextWordPair();
+
+        if (currentPair != null)
+        {
+            UpdateText(currentPair[0], currentPair[1]);
+        }
```

Comment:
Nit: add a log message to indciate that the next word pair was null.

Reply:
currentPair == null indicates there doesn't need to be an update to the text (at the end or start of vocab list dependent on prevCard or nextCard being run) so an error message is needed but can add a regular log to console if thats what you want?

Reply:
Yeah a log message just to indicate that we reached either end of the list might not be super useful, since we will know when it doesn't switch. I'll leave it up to you.